### PR TITLE
feat: playlist management

### PR DESCRIPTION
**Pull request by maitrakhatri**
# features
- add and remove videos from Playlist from Homepage, video listing page and single video page

### CODE REVIEW

File: src/Components/AddToPlaylistModal.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <div className="add-to-playlist">
+      <h6>Add to Playlist</h6>
+      <img
+        src="../../assets/icons/close.svg"
+        id="close"
+        alt="close"
+        onClick={() => setShowModal(false)}
+      />
```
Comment (devananth) [inline src/Components/AddToPlaylistModal.jsx:48]: close icon is not visible, kindly check it once

**Review by devananth — APPROVED** [review]
Everything else good,

File: src/Components/AddToPlaylistModal.jsx (hunk 1 above)
Comment (maitrakhatri) [reply src/Components/AddToPlaylistModal.jsx:48]: yes I added the icon

**Review by anupkgurung — APPROVED** [review]
Clean code.
Functionality is smooth

File: src/Context/playlist-context.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                }
+            });
+            setMyPlaylists(() => myPlaylists.map((playlist) => playlist._id === playlistId ? response.data.playlist : playlist))
+        }
+        catch (error) {
+            console.log(error)
+        }
+    }
+
+    return <PlaylistContext.Provider value={{ getAllPlaylists, createNewPlaylist, deletePlaylist, getSpecificPlaylists, addVideoToPlaylist, deleteVideoFromPlaylist, myPlaylists, showModal, setShowModal, newPlaylist, setNewPlaylist, targetVideo, setTargetVideo, specificPlaylist }}>
```
Comment (MarkVed17) [inline src/Context/playlist-context.js:104]: I specifically like how all the states are managed here. Reduces the overhead of maintaining multiple states on different pages.

**Review by MarkVed17 — APPROVED** [review]
Rest all looks good.
You may proceed.

File: src/Context/playlist-context.js (hunk 2 above)
Comment (maitrakhatri) [reply src/Context/playlist-context.js:104]: Thanks !!

**Pull request merged**